pkg/client/intermediate/intermediate.go

```diff
@@ -0,0 +1,139 @@
+package intermediate
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+
+	"github.com/bitnami-labs/charts-syncer/pkg/client"
+
+	"github.com/juju/errors"
+
+	"github.com/bitnami-labs/charts-syncer/internal/utils"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/types"
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+var (
+	versionRe = regexp.MustCompile("(.*)-(\\d+\\.\\d+\\.\\d+)\\.bundle.tar")
+)
+
+// Repo allows to operate a chart repository.
+type Repo struct {
```

Comment:
Ideally, I would treat the intermediate "store" as a new object in the configuration:

```yaml
source:
  intermediate:
    kind: BUNDLES
    path: /path/to/bundles/dir
```

This way we should have `pkg/client/intermediate/intermediate.go` including a generic `New` constructor and then a `pkg/client/intermediate/bundles` with this logic. However, I am totally fine having this simplification today.

Reply:
I would skip it for now